src/batou/secrets/encryption.py

```diff
@@ -433,3 +448,296 @@ def age(cls):
+            except OSError as err:
+                if err.errno == errno.EIO:
+                    # work arond suspected pty "feature", where
+                    # reading from the file descriptor
+                    # when there is no data raises instead of
+                    # returning an empty string, see
+                    # https://bugs.python.org/issue5380
+                    chunk = None
+            if not chunk:
+                break
+            buffer += chunk
+
+        # Wait for the child to exit
+        pid, exitcode = os.waitpid(child_pid, 0)
+
+        if exitcode != 0 or pid != child_pid:
+            # print stderr, stdout
+            buf_stderr = b""
+            while True:
+                try:
+                    chunk = os.read(stderr_r, 1024)
+                except OSError as err:
+                    if err.errno == errno.EIO:
+                        # work arond suspected pty "feature", where
+                        # reading from the file descriptor
+                        # when there is no data raises instead of
+                        # returning an empty string, see
+                        # https://bugs.python.org/issue5380
+                        chunk = None
+                if not chunk:
+                    break
+                buf_stderr += chunk
+
+            raise AgeCallError.from_context(args, exitcode, buf_stderr)
+
+        os.close(fd)
+
+        return buffer.decode("utf-8").replace("\n", "\\n")
+
+    def decrypt(self):
+        # read the entire file, parse as ConfigUpdater
+
+        if not self.locked:
+            raise ValueError("File is not locked")
+
+        config_encrypted = ConfigUpdater().read(self.path)
+        config = ConfigUpdater().read(self.path)
+
+        # for each section
+        for section in config.sections():
+            # if section is called batou, skip
+            if section == "batou":
+                continue
+            # for each option
+            for option in config[section]:
+                # decrypt the value
+                config[section][option].value = self.decrypt_age_string(
+                    config[section][option].value
+                )
+
+        # cache the decrypted content
+        self._decrypted_content = config
+        self._encrypted_content = config_encrypted
+
+        # return the decrypted content as bytes
+        return str(config).encode("utf-8")
+
+    def _write(
+        self, content: bytes, recipients: List[str], reencrypt: bool = False
+    ):
+        # parse the content as ConfigUpdater
+        config = ConfigUpdater()
+        config.read_string(content.decode("utf-8"))
+
+        # for each section
+        for section in config.sections():
+            # if section is called batou, skip
+            if section == "batou":
+                continue
+            # for each option
+            for option in config[section]:
+                if (
+                    not reencrypt
```

Comment:
This is hard to read, I suggest two changes: a) make the happy path go first and b) extract the conditions into easier-to-read variables:

```python
new_value = config[section][option].value
assert new_value is not None

try:
    old_value = self._decrypted_content[section][option].value
except Exception:
    old_value = None

value_has_changed = new_value != old_value

if (reencrypt or value_has_changed):
    new_encrypted_value = self.encrypt_age_string(new_value, recipients)
else:
    new_encrypted_value = self._encrypted_content[section][option].value

config[section][option].value = new_encrypted_value
```